### ISSUE TITLE
FR: Differentiate between displayed and inserted text in Custom dictionaries

### ISSUE THREAD

**Issue by chrisgrieser** [tadashi-aikawa/obsidian-various-complements-plugin#104]
Basically, what I am suggesting is an improvement to #37.
So in the the custom dictionary, we have currently three types of text:
1. text that will be inserted, *and* is displayed in the suggester, *and* is matched (= insert-display-matching text)
2. Text that only be *displayed* in the suggester as a subtitle (= description text)
3. additional text that is matched(= matching text)

with the [Delimiter to hide a suggestion](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.6.%20Custom%20dictionary%20complement/%E2%9A%99%EF%B8%8FDelimiter%20to%20hide%20a%20suggestion/) we kind of get a fourth type of text:
4. text that will be inserted, but is *not* displayed in the suggester. (= display text)

---

I am envisioning something like this, basically making 1) display-only-text
1. text that is *only* displayed in the suggester (= display text)
2. text that is only displayed in the suggester as subtitle (= description text)
3. text that is matched (= matching text)
4. text that will be inserted (= insert text)

This would improve things like the [citation autocompleter I built using your plugin](https://discord.com/channels/686053708261228577/722584061087842365/953301880291876936), but it would also increase the flexibility of Various Complements in general for all kinds of use cases.

**Comment by tadashi-aikawa** [comment]
Hi, @chrisgrieser. Thanks for your FR :)

If it's ok with you that the specifications are as follows, I could implement them quickly.

```
// dictionary file. ;;;; is delimiter for this feature
displayed-text;;;;<div>template</div>,description,aliases1,aliases2
```

- It shows `displayed-text`, not `<div>template</div>`
- It inserts `<div>template</div>`, not `display-text`
- It matches by search both `displayed` and `template`

**Comment by chrisgrieser** [comment]
> - It shows displayed-text, not <`div>template</div>`
> - It inserts `<div>template</div>`, not display-text
> - It matches by search both displayed and template

yes, that would suffice, too!

**Comment by tadashi-aikawa** [comment]
@chrisgrieser 

I have updated beta version to [obsidian-various-complements-plugin-beta v5.10.0-beta](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/5.10.0.beta) 🚀 

Please confirm whether it works as you expected. (and hopefully not found bugs 😉 )

![](https://user-images.githubusercontent.com/9500018/159123341-ee1919c1-385f-4e12-b0ee-83163433fbbf.gif)

**Comment by tadashi-aikawa** [comment]
Released [v5.10.0.beta-2](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/5.10.0.beta2) 🚀

**Comment by chrisgrieser** [comment]
can confirm, works as intended :)

**Comment by tadashi-aikawa** [comment]
Thanks for your confirmation!

I will reopen this issue because I have not released it officially :)

And I will modify [🖼️Add a word to a custom dictionary dialog](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/5.%20Terms/%F0%9F%96%BC%EF%B8%8FAdd%20a%20word%20to%20a%20custom%20dictionary%20dialog/), too 👍

**Comment by tadashi-aikawa** [comment]
P.S. This feature is very compatible with like callouts 😄 

![image](https://user-images.githubusercontent.com/9500018/159163645-2aa3cc54-ddd7-4a49-9a18-4f47b3f2aeb5.png)

**Comment by chrisgrieser** [comment]
> P.S. This feature is very compatible with like callouts 😄

that was **exactly** what I was thinking and doing right now! I added this to my custom dictionary (`|` as delimiter, `>>>` as insert/display delimiter)

```text
%% Callouts %%
note>>>> [!NOTE]\n> |callout|note|callout
abstract>>>> [!ABSTRACT]\n> |callout|abstract|callout
summary>>>> [!SUMMARY]\n> |callout|summary|callout
tldr>>>> [!TLDR]\n> |callout|tldr|callout
info>>>> [!INFO]\n> |callout|info|callout
todo>>>> [!TODO]\n> |callout|todo|callout
tip>>>> [!TIP]\n> |callout|tip|callout
hint>>>> [!HINT]\n> |callout|hint|callout
important>>>> [!IMPORTANT]\n> |callout|important|callout
success>>>> [!SUCCESS]\n> |callout|success|callout
check>>>> [!CHECK]\n> |callout|check|callout
done>>>> [!DONE]\n> |callout|done|callout
question>>>> [!QUESTION]\n> |callout|question|callout
help>>>> [!HELP]\n> |callout|help|callout
faq>>>> [!FAQ]\n> |callout|faq|callout
warning>>>> [!WARNING]\n> |callout|warning|callout
caution>>>> [!CAUTION]\n> |callout|caution|callout
attention>>>> [!ATTENTION]\n> |callout|attention|callout
failure>>>> [!FAILURE]\n> |callout|failure|callout
fail>>>> [!FAIL]\n> |callout|fail|callout
missing>>>> [!MISSING]\n> |callout|missing|callout
danger>>>> [!DANGER]\n> |callout|danger|callout
error>>>> [!ERROR]\n> |callout|error|callout
bug>>>> [!BUG]\n> |callout|bug|callout
example>>>> [!EXAMPLE]\n> |callout|example|callout
quote>>>> [!QUOTE]\n> |callout|quote|callout
cite>>>> [!CITE]\n> |callout|cite|callout
```

**Comment by tadashi-aikawa** [comment]
We were thinking about the same thing!

At that time, I thought it was satisfied using aliases without implementing the new feature. But I noticed **we need not memorize the list of callouts and can filter even after inputting words** if there are the same prefixes (ex: ad-), and it can't be realizable without your feature request 👍 

```
📝ad-note >>> > [!note]\n> ,メモ書き
ℹ️ad-info >>> > [!info]\n> ,関連情報
🔥ad-hint >>> > [!hint]- <CARET>\n> ,深堀りした情報(タイトル: 深堀する内容)
⚠ad-attention >>> > [!attention]\n> ,見なくてもいい注意書き
⚠ad-caution >>> > [!caution]\n> ,被害を被る可能性がある、見てほしい注意書き
⚠ad-warning >>> > [!warning]\n> ,被害を被る可能性が高いので、必ず見てほしい注意書き
🌩ad-danger >>> > [!danger]\n> ,必ず被害を被るので、絶対に見てほしい注意書き
🌩ad-error >>> > [!error]- <CARET>\n> ,エラーが発生したときの補足(タイトル: エラー文言)
```

**Comment by tadashi-aikawa** [comment]
Coming soon! 🦉 

https://user-images.githubusercontent.com/9500018/159889512-dec2a1d5-96f0-4635-9fb1-dd785b8ce62e.mp4

**Comment by tadashi-aikawa** [comment]
https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/5.10.0.beta3

**Comment by tadashi-aikawa** [comment]
Released [v5.10.0](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/5.10.0) 🚀

**Comment by chrisgrieser** [comment]
Btw, you should definitely showcase using Various complements for callouts. It's in my view the best method to write callouts, and it works be a bummer when only the two of us do that 😆 

In the docs and Discord or sth?

**Comment by tadashi-aikawa** [comment]
Sounds good!

I'll create a page in the docs and some custom dictionary example files first.

**Comment by tadashi-aikawa** [comment]
Added https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/6.%20Showcase/%F0%9F%93%93Custom%20dictionary%20examples/
